examples/sdk-upload/README.md

```diff
@@ -0,0 +1,334 @@
+var path = require('path')
+
+// assign our region from our environment variables
+AWS.config.region = process.env.S3_REGION
+
+/**
+* Returns data specific to our upload that we send to the front end
+* @param {Buffer} file - file that we are uploading
+* @param {string} filename - name of the file to be uploaded
+* @param {function} callback
+**/
+function upload (file, filename, callback) {
+  // creating our new filename
+  var filenameHex =
+  filename.split('.')[0] +
+  crypto.randomBytes(8).toString('hex') +
+  path.extname(filename)
+  // loading our bucket name from our environment variables
+  var bucket = process.env.S3_BUCKET
+  // creating a new instance of our bucket
+  var s3Bucket = new AWS.S3({params: {Bucket: bucket}})
+  // sets the params needed for the upload
+  var params = {Bucket: bucket, Key: filenameHex, Body: file}
+  // SDK upload to S3
+  s3Bucket.upload(params, function (err, data) {
+    if (err) console.log(err, data)
+    // callback with the data that gets returned from S3
+    else callback(null, data)
+  })
+}
+
+module.exports = {
+  upload
+}
+```
+
+#### We've now set up our S3 upload function!
+
+
+### Step 5 - Create a server to handle our upload file
+
+Create a directory called `lib`. This will hold our server + routes.
+
+Create a file called `index.js` and add the following:
+
+```js
+'use strict'
+
+var Hapi = require('hapi')
+var Inert = require('inert')
+var assert = require('assert')
+
+// creating a new instance of our server
+var server = new Hapi.Server()
+
+// connecting to port 8000
+server.connection({port: 8000})
+// registering the Inert module which allows us to render static files
+server.register([Inert], function (err) {
+  assert(!err) // not much point continuing without plugins ...
+
+  // adding our routes (we'll be creating this file next)
+  server.route(require('./routes.js'))
+
+  // starting the server
+  server.start(function (err) {
+    assert(!err) // not much point continuing if the server does not start ...
+    console.log('The server is running on: ', server.info.uri)
+  })
+})
+
+module.exports = server
+```
+
+Next create a file called `routes.js` and add the following:
+
+```js
+// require the function we just created
+var s3 = require('../src/upload.js')
+// path is needed to resolve the filepath to our index.html
+var path = require('path')
+
+module.exports = [
+  // our index route
+  {
+    method: 'GET',
+    path: '/',
+    handler: function (request, reply) {
+      return reply.file(path.resolve(__dirname, '../public/index.html'))
+    }
+  },
+  // our endpoint
+  {
+    method: 'POST',
+    path: '/file_submitted',
+    handler: function (request, reply) {
+      // we'll be sending through a file along with it's filename
+      var file = request.payload.file
+      var filename = request.payload.filename
+      // upload the file to S3
+      s3.upload(file, filename, function (err, data) {
+        if (err) console.log(err)
```

Comment:
here you can do `request.handleError(err, err);`
(_instead of having an `if` statement and having to write a another test for it..._)